test/cql-pytest/test_describe.py

```diff
@@ -447,6 +447,75 @@ def test_desc_index(cql, test_keyspace):
+            ("idx_f",           f"CREATE INDEX idx_f ON {tbl}(values(f))"),
+        ]
+
+        for name, create_stmt in indexes:
+            cql.execute(create_stmt)
+            desc = cql.execute(f"DESC INDEX {test_keyspace}.{name}").one().create_statement
+            assert create_stmt in desc or create_stmt.replace("(", " (", 1)
+        
+        # When an index is created on list column with `CREATE INDEX ON tbl(list_column)`,
+        # its target is `values(list_column)` and describe will print creat statement with that
+        # values() function added.
+        cql.execute(f"CREATE INDEX idx_g ON {tbl}(g)")
+        idx_g_desc = cql.execute(f"DESC INDEX {test_keyspace}.idx_g").one().create_statement
+        idx_g_create_stmt = f"CREATE INDEX idx_g ON {tbl}(values(g))"
+        assert idx_g_create_stmt in idx_g_desc or idx_g_create_stmt.replace("(", " (", 1)
+
+        # The same situation but with set
+        cql.execute(f"CREATE INDEX idx_h ON {tbl}(h)")
+        idx_h_desc = cql.execute(f"DESC INDEX {test_keyspace}.idx_h").one().create_statement
+        idx_h_create_stmt = f"CREATE INDEX idx_h ON {tbl}(values(h))"
+        assert idx_h_create_stmt in idx_h_desc or idx_h_create_stmt.replace("(", " (", 1)
+
+def test_desc_index_column_quoting(cql, test_keyspace):
+    # In this test, all assertions are in form of
+    # `assert create_stmt in desc or create_stmt.replace("(", " (", 1)`
+    # This is because Scylla and Cassandra have slightly different (whitespace-wise)
+    # output of `DESC INDEX`. 
+    #   Scylla:     `CREATE INDEX ON table_name(column_name)`
+    #   Cassandra:  `CREATE INDEX ON table_name (column_name)`
+
+    quoted_names = ['"select"', '"hEllo"', '"x y"', '"hi""hello""yo"', '"""hi"""', '"keys(m)"', '"values(m)"', '"entries(m)"']
+    with new_test_table(cql, test_keyspace, "a int primary key, \"some???name\" map<int, int>, " + ', '.join([name + " int" for name in quoted_names])) as tbl:
+        indexes = [
+            ("idx_1", f"CREATE INDEX idx_1 ON {tbl}(keys(\"some???name\"))"),
+        ] + [(f"idx_{n}", f"CREATE INDEX idx_{n} ON {tbl}({name})") for n, name in enumerate(quoted_names, 2)]
+
+        for name, create_stmt in indexes:
+            cql.execute(create_stmt)
+            desc = cql.execute(f"DESC INDEX {test_keyspace}.{name}").one().create_statement
+            assert create_stmt in desc or create_stmt.replace("(", " (", 1)
+
+def test_desc_local_secondary_index(scylla_only, cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int primary key, b int, c frozen<list<int>>") as tbl:
+        indexes = [
+            ("idx_1", f"CREATE INDEX idx_1 ON {tbl}((a), b)"),
+            ("idx_2", f"CREATE INDEX idx_2 ON {tbl}((a), full(c))"),
+        ]
+        
+        for name, create_stmt in indexes:
+            cql.execute(create_stmt)
+            desc = cql.execute(f"DESC INDEX {test_keyspace}.{name}").one().create_statement
+            assert create_stmt in desc
```

Comment:
Why is this "in" and not equal?

Reply:
I was doing assertions this way in the whole test file. I could also do `assert create_stmt == desc.trim()` in most places but I just left it this way.